main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -214,7 +214,11 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
 
     def Class: Rule1[ParsedAst.Declaration] = {
       def Head = rule {
-        Documentation ~ Modifiers ~ SP ~ keyword("class") ~ WS ~ Names.Class ~ optWS ~ "[" ~ optWS ~ ConstrainedTypeParam ~ optWS ~ "]"
+        Documentation ~ Modifiers ~ SP ~ keyword("class") ~ WS ~ Names.Class ~ optWS ~ "[" ~ optWS ~ ConstrainedTypeParam ~ optWS ~ "]" ~ OptSuperClasses
+      }
+
+      def OptSuperClasses = rule {
+        optional(optWS ~ keyword("extends") ~ optWS ~ "[" ~ optWS ~ oneOrMore(Names.QualifiedClass).separatedBy(optWS ~ "," ~ optWS) ~ optWS ~ "]") ~> ((o: Option[Seq[Name.QName]]) => o.getOrElse(Seq.empty))
```

Comment:
`extends` is meant as a placeholder until the syntax is locked in. I chose it since it isn't used anywhere else (easier for semi-automated changes) and because it kind of matches the other placeholder `with` syntax. I think once we agree on a more permanent syntax, we can make the change all at once.

> `Eq[a]`

Should I add checks to ensure the variable name matches?

Reply:
OK and yes.